### DISCUSSION TITLE
Support freedesktop.org specifications

### DISCUSSION THREAD

**Discussion by dcpurton** [neomutt/neomutt#3287] [Development]
Issue #3278 raises questions of whether and to what extent NeoMutt should provide files supporting the freedesktop.org specs.

This discussion thread provides some relevant specification links, content ideas and place for discussion.

## Specifications:

- [Desktop Entry Specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html)
- [AppStream Upstream Metadata](https://freedesktop.org/software/appstream/docs/chap-Metadata.html#sect-Metadata-GenericComponent)
- [AppStream Metadata Quickstart](https://freedesktop.org/software/appstream/docs/chap-Quickstart.html)

## To support the specification NeoMutt would need to:

- Provide a `.desktop` file (installed as `/usr/share/applications/org.neomutt.NeoMutt.desktop`)
- Provide at least one screenshot
- Provide a component metadata file (installed as `/usr/share/metainfo/org.neomutt.NeoMutt.metainfo.xml`)
- Install existing `contrib/logo/` files in `/usr/share/icons/hicolor/<size>/apps/`

## Considerations:

- Which OS support the freedesktop.org standards?
- What content should be supplied for descriptions?
- What translations should be supplied for the various descriptions?
- What are the ongoing efforts required for each release and who will maintain things?

## Why bother?

It's true that the freedesktop.org specifications are largely geared towards GUI applications, but specifications work best for a user when everyone plays along, so perhaps it's worth considering.

**Comment by dcpurton** [comment]
## `.desktop` file

### src location

```
contrib/org.neomutt.NeoMutt.desktop
```

### install location

```
/usr/share/applications/org.neomutt.NeoMutt.desktop
```

### Suggested content

```desktop
[Desktop Entry]
Type=Application
Version=1.5
Name=NeoMutt
GenericName=Mail User Agent
NoDisplay=false
Comment=A text-based Mail User Agent
Icon=neomutt
Exec=neomutt %u
Terminal=true
MimeType=message/rfc822;x-scheme-handler/mailto;
Categories=Network;Email;News;ConsoleOnly;
Keywords=email;e-mail;news;smtp;pop;imap;nntp;maildir;mbox;
StartupNotify=false
```

**Comment by dcpurton** [comment]
## Screenshots

These screenshots are taken using `contrib/samples/sample.neomuttrc` with a 16×9 aspect ratio. The INBOX is the January 2022 neommutt-users mailing list archive.

Note: Screenshots need not be installed.

### src location

```
contrib/screenshots/
```

### Default screenshot

```
contrib/screenshots/neomutt-screenshot-index.png
```

![neomutt-screenshot-index](https://user-images.githubusercontent.com/6268044/156870772-db8c2771-3bee-4372-975b-20cd98dc174a.png)

### Secondary screenshot

```
contrib/screenshots/neomutt-screenshot-pager.png
```

![neomutt-screenshot-pager](https://user-images.githubusercontent.com/6268044/156870800-c4b77651-8ca6-47b3-9028-5b69dabdbaa4.png)

**Reply by flatcap** [reply]
Your second screenshot is a bit weird.
It looks like it's a transparent window on top of the first screenshot.
You can see the tree characters bleeding though into the pager area.

**Reply by flatcap** [reply]
Ages ago, I created screenshots (with dummy data) for most dialogs: [Screenshots](https://github.com/neomutt/gfx/blob/main/screenshots/screenshot/README.md#screenshots-of-neomutt)
They come in 80x24 and 100x30 variants (png and svg).

**Reply by flatcap** [reply]
The screenshots need to be "**achievable**".

Out-of-the-box, NeoMutt looks very dull.
With a few colours, some emoji chars and/or Nerd Fonts we can make NeoMutt look pretty.

I think that whatever we show in the screenshots must be easily reproducable by a beginner.
We should ship the config/theme in `contrib/`

**Reply by flatcap** [reply]
We want some screenshots to really sell NeoMutt.

- Index + Sidebar
  Colours, characters

- Index + Pager + Sidebar
  Highlighting in the Pager, e.g. a diff

- Compose
  lots of addressees, multipart attachments

- Address book (aliases)
  Several tagged entries

Can we add a title to each dialog?
(so that we can tell the view _what_ they're looking at)

What would be the best way to showcase NotMuch?

**Reply by dcpurton** [reply]
> Your second screenshot is a bit weird.
> It looks like it's a transparent window on top of the first screenshot.
> You can see the tree characters bleeding though into the pager area.

Indeed it is. I forgot to put the black background behind. My terminal does have a slightly transparent background.

**Reply by dcpurton** [reply]
> The screenshots need to be "**achievable**".
> 
> Out-of-the-box, NeoMutt looks very dull.
> With a few colours, some emoji chars and/or Nerd Fonts we can make NeoMutt look pretty.
> 
> I think that whatever we show in the screenshots must be easily reproducable by a beginner.
> We should ship the config/theme in `contrib/`

Yes, this is why I used `contrib/sample.neomuttrc`.

But this configuration isn't that exciting.

**Reply by dcpurton** [reply]
> Ages ago, I created screenshots (with dummy data) for most dialogs: [Screenshots](https://github.com/neomutt/gfx/blob/main/screenshots/screenshot/README.md#screenshots-of-neomutt)
> They come in 80x24 and 100x30 variants (png and svg).

That's a useful resource for type of screenshots. For the purposes of the metadata file it probably only needs a few screenshots though, not every possible one.

The spec suggests using a 16×9 ratio for screen shots with default window decorations (something harder than you imagine when you use a tiling window manager with no window decorations enabled 🙂)

**Reply by dcpurton** [reply]
> We want some screenshots to really sell NeoMutt.
> 
> - Index + Sidebar
>   Colours, characters
> 
> - Index + Pager + Sidebar
>   Highlighting in the Pager, e.g. a diff
> 
> - Compose
>   lots of addressees, multipart attachments
> 
> - Address book (aliases)
>   Several tagged entries
> 
> Can we add a title to each dialog?
> (so that we can tell the view _what_ they're looking at)

The metadata file allows for captions.

> 
> What would be the best way to showcase NotMuch?

I confess I don't use NotMuch. But maybe I should?

**Reply by dcpurton** [reply]
> We want some screenshots to really sell NeoMutt.

Perhaps you could have a competition:

## Submit your best default screenshot and self contained theme to sell NeoMutt!

- Submit `name-theme.neomuttrc` to be included in the `~/theme-test.neomuttrc` below.
- Submit a default screen shot which includes the index.
- Submit up to 4 other screenshots showing (e.g., pager, attachment lists, compose window, NotMuch integration, etc.
- NeoMutt should be run as `neomutt -n -F theme-test.neomuttrc`.

``` ini
source "~/neomutt-theme-competition/name-theme.neomuttrc"

unset record
unset signature

set folder = "~/neomutt-theme-competition"
set from = "john.doe@example.com"
set realname = "John Doe"

set spoolfile = "+INBOX"

# set other features you definitely want to showcase, e.g., set sidebar_visible.

# vim: syn=neomuttrc
```

### Screenshot specifications

- Screenshot aspect ration should be 16×9.
- Terminal window should be between 80 and 100 characters wide.

---

You could provide `INBOX` (and other files to showcase the side bar) as suitable sample `mbox` files. You would want a range of emails that looked real and included PGP signed/encrypted emails, emails with attachments, threads, emojis in subject lines, multipart emails, etc.

The screenshots don't have to be in their final format, since the output should be reproducible and you can makes a series that all match once you settle on a final sample config.

**Reply by dcpurton** [reply]
Here's a starting point for setting up a colour theme to submit screenshots.

This is similar to the config for one of my accounts except I don't use the sidebar or have the index visible while view messages.

To test unpack attached `dcpurton.tar.gz` and run:

```
neomutt -n -F theme-test.neomuttrc
```

## `theme-test.neomuttrc`

```ini
unset signature

set folder = "."
set from = "john.doe@example.com"
set mail_check_stats = yes
set postponed = "+Drafts"
set realname = "John Doe"
set read_only = yes
set record = "+Sent"
set sidebar_format = "%D%?F? [%F]?%* %?N?%N/?%S "
set sidebar_visible = yes
set sidebar_width = 20
set spoolfile = "+INBOX"
set trash = "+Bin"

named-mailboxes INBOX +INBOX Sent +Sent Archive +Archive Drafts +Drafts Bin +Bin

bind index,pager \Cj sidebar-next
bind index,pager \Ck sidebar-prev
bind index,pager \Co sidebar-open
bind index,pager \Cv sidebar-toggle-visible

# source theme file
source "./dcpurton-theme.neomuttrc"

# vim: syn=neomuttrc
```

## Screenshot

![dcpurton-screenshot](https://user-images.githubusercontent.com/6268044/157052296-b4452b7c-8db8-43f9-bc0f-ca1959c4203a.png)

## Config and test files

[dcpurton.tar.gz](https://github.com/neomutt/neomutt/files/8198369/dcpurton.tar.gz)

`INBOX` contains a few messages with varying structure.

**Comment by dcpurton** [comment]
## Upstream metadata

### src location

```
contrib/org.neomutt.NeoMutt.metainfo.xml
```

### install location

```
/usr/share/metainfo/org.neomutt.NeoMutt.metainfo.xml
```

### suggested content

Notes:

- The description below is taken from the Debian package. It probably needs to be re-written.
- The release info is taken from the most recent release. It's probably not worth filing in NeoMutt's complete release history.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<component type="desktop-application">
  <id>org.neomutt.NeoMutt</id>
  <metadata_license>CC0-1.0</metadata_license>
  <project_license>GPL-2.0+</project_license>
  <name>NeoMutt</name>
  <summary>A text-based Mail User Agent</summary>
  <icon type="remote" width="256" height="256">https://github.com/neomutt/neomutt/raw/master/contrib/logo/neomutt-256.png</icon>

  <description>
    <p>
      NeoMutt is based on Mutt but with added features. It supports all the
      features that are supported by Mutt, including POP and IMAP support, SSL
      encryption and SASL authentication, threading and GPG support.
    </p>
    <p>
      On the top of that, Neomutt provides:
    </p>
    <ul>
      <li>Notmuch: a mail indexing tools that provides advanced features such as
        full-text search, thread reconstruction and added features.</li>
      <li>Color attachment headers using regex, just like mail bodies.</li>
      <li>Custom rules for theming the mail index.</li>
      <li>NNTP support.</li>
      <li>Visual progress bar for slow operations.</li>
      <li>Trash folder.</li>
    </ul>
  </description>

  <categories>
    <category>Network</category>
    <category>Email</category>
    <category>News</category>
    <category>ConsoleOnly</category>
  </categories>

  <provides>
    <binary>neomutt</binary>
  </provides>

  <launchable type="desktop-id">org.neomutt.NeoMutt.desktop</launchable>

  <screenshots>
    <screenshot type="default">
      <caption>Mail index</caption>
      <image>https://github.com/neomutt/neomutt/raw/master/contrib/screenshots/neomutt-screenshot-index.png</image>
    </screenshot>
    <screenshot>
      <caption>Mail index and pager</caption>
      <image>https://github.com/neomutt/neomutt/raw/master/contrib/screenshots/neomutt-screenshot-pager.png</image>
    </screenshot>
  </screenshots>

  <content_rating type="oars-1.1">
    <content_attribute id="social-chat">intense</content_attribute>
    <content_attribute id="social-contacts">intense</content_attribute>
  </content_rating>  

  <url type="homepage">https://neomutt.org/</url>
  <url type="bugtracker">https://github.com/neomutt/neomutt/issues</url>
  <url type="help">https://neomutt.org/guide/</url>
  <url type="donation">https://www.paypal.me/russon/</url>
  <url type="translate">https://neomutt.org/translate</url>
  <url type="contact">https://neomutt.org/about#contact-the-developers</url>

  <developer_name>Neomutt Developers</developer_name>
  <update_contact>neomutt-devel@neomutt.org</update_contact>

  <releases>
    <release version="20211029" date="2021-10-29">
      <description>
        <p>This is a bug-fix release.</p>
        <p>Features:</p>
        <ul>
          <li>Notmuch: support separate database and mail roots without .notmuch</li>
        </ul>
        <p>Bug Fixes:</p>
        <ul>
          <li>fix notmuch crash on open failure</li>
          <li>fix crypto crash handling pgp keys</li>
          <li>fix ncrypt/pgp file_get_size return check</li>
          <li>fix restore case-insensitive header sort</li>
          <li>fix pager redrawing of long lines</li>
          <li>fix notmuch: check database dir for xapian dir</li>
          <li>fix notmuch: update index count after <code>&lt;entire-thread&gt;</code></li>
          <li>fix protect hash table against empty keys</li>
          <li>fix prevent real_subj being set but empty</li>
          <li>fix leak when saving fcc</li>
          <li>fix leak after <code>&lt;edit-or-view-raw-message&gt;</code></li>
          <li>fix leak after trash to hidden mailbox</li>
          <li>fix leak restoring postponed emails</li>
        </ul>
      </description>
    </release>
  </releases>
</component>
```

**Reply by flatcap** [reply]
Describing the NeoMutt -- It's something I've been putting off for years!
Once we figure out the wording it needs to get copied everywhere: docs, website, github, etc

**Reply by flatcap** [reply]
> It's probably not worth filing in NeoMutt's complete release history

True, but we could abbreviate it.
One app contains a list of empty, but dated releases:
```
<release version="2.8.22" date="2017-05-11"/>
<release version="2.8.20" date="2017-02-01"/>
```

We could do that for a handful of recent releases,
but I favour one addition: a URL to the GitHub Release
```
<release version="2021-10-29" type="development" date="2021-10-29">
  <url type="details">https://github.com/neomutt/neomutt/releases/tag/20211029</url>
</release>
```

**Reply by dcpurton** [reply]
> > It's probably not worth filing in NeoMutt's complete release history
> 
> True, but we could abbreviate it.
> One app contains a list of empty, but dated releases:
> ```
> <release version="2.8.22" date="2017-05-11"/>
> <release version="2.8.20" date="2017-02-01"/>
> ```
> 

I notice a number of packages do this kind of thing.

> We could do that for a handful of recent releases,
> but I favour one addition: a URL to the GitHub Release
> ```
> <release version="2021-10-29" type="development" date="2021-10-29">
>   <url type="details">https://github.com/neomutt/neomutt/releases/tag/20211029</url>
> </release>
> ```
> 

That's a good idea. You could just keep the detail inline for the most recent release.

**Comment by vuori** [comment]
We talked about this a bit on IRC, and the suggestions here look fine. Would you be willing to put together a PR for this?
